packages/core/src/model.ts

```diff
@@ -66,6 +67,21 @@ export function createModel(
         'context' in config ? config : { ...config, context: initialContext },
         implementations
       );
+    },
+    withInput: (input: any) => {
+      const model = createModel(initialContext, creators as any);
```

Comment:
we should be able to just spread the existing `model` from the outer scope, right? any particular reason that we recreate the model here?